drivers/wifi/eswifi/eswifi_shell.c

```diff
@@ -39,7 +40,7 @@ static int eswifi_shell_atcmd(const struct shell *sh, size_t argc,
 	eswifi_lock(eswifi);
 
 	memset(eswifi->buf, 0, sizeof(eswifi->buf));
-	for (i = 1; i < argc; i++) {
+	for (i = 1; i < MIN(argc, sizeof(eswifi->buf)); i++) {
```

Comment:
This doesn't really help here, as we can still overflow the buf. We would need to keep track of how big the buffer is, and use something that can cat without doing an overflow. This isn't strncat, which wouldn't terminate it.

Reply:
gosh ... thanks we have code review. Have written without look it properly.